test/reduce/reduce_with_index.cpp

```diff
@@ -1,248 +1,163 @@
 // SPDX-License-Identifier: MIT
-// Copyright (c) 2018-2023, Advanced Micro Devices, Inc. All rights reserved.
+// Copyright (c) 2018-2024, Advanced Micro Devices, Inc. All rights reserved.
 
 #include <getopt.h>
 
 #include "ck/library/utility/host_common_util.hpp"
 #include "profiler/profile_reduce_impl.hpp"
-
+#include <gtest/gtest.h>
 using namespace ck;
 
-static struct option long_options[] = {{"inLengths", required_argument, nullptr, 'D'},
-                                       {"reduceDimensions", required_argument, nullptr, 'R'},
-                                       {"scales", required_argument, nullptr, 'S'},
-                                       {"help", no_argument, nullptr, '?'},
-                                       {nullptr, 0, nullptr, 0}};
-
-class SimpleAppArgs
+struct ReduceParam
 {
-    private:
-    int option_index = 0;
-
-    public:
+    bool do_verification;
+    bool propagateNan;
+    bool useIndex;
+    bool time_kernel;
+    bool do_dumpout;
+    int init_method;
+    float alpha;
+    float beta;
     std::vector<size_t> inLengths;
     std::vector<int> reduceDims;
-    std::vector<float> scales;
+    ReduceTensorOp reduceOpId;
+};
 
-    int data_type;
-    int init_method = 1;
+std::vector<std::vector<int>> settGenericReduceDim()
+{
+    return {{0, 1, 2, 3}, {0, 1, 2}, {0, 1, 3}, {0, 2, 3}, {1, 2, 3}, {0}, {1}, {2}, {3}};
+}
 
-    public:
-    void show_usage(const char* cmd)
-    {
-        std::cout << "Usage of " << cmd << std::endl;
-        std::cout << "--inLengths or -D, comma separated list of input tensor dimension lengths "
-                     "(only 4-d tensor supported)"
-                  << std::endl;
-        std::cout << "--reduceDimensions or -R comma seperated list of dimension indexes to reduce "
-                     "(only 1 or 3 or 4 dimensions supported)"
-                  << std::endl;
-        std::cout << "--scales or -S, comma separated two float values for alpha and beta"
-                  << std::endl;
-        std::cout << "Arg1 -- data type (1: fp32, 3: int8, 5: bp16, 6: fp64)" << std::endl;
-        std::cout << "Arg2 -- init method(0=no init, 1=single integer value, 2=scope integer "
-                     "value, 3=decimal value)"
-                  << std::endl;
-    };
+ReduceParam sett0()
```

Comment:
Maybe rename like this?
```suggestion
ReduceParam setDefault()
```

Reply:
You can just remove this function and add default values in the structure.

Reply:
I agree with it. However, it is a conscious exclusion. In the next steps the following structure will be moved to a reduce file to resolve implicit calls.

Reply:
So why you doSo why not move it now?

Reply:
The pull-request contains only changes in this test: one case, one PR. Changes in structure interferes in API.

Reply:
Sorry but I dont follow - why structure for test cases parameter require API changes? Maybe we can sync offline